docs/doc.md

```diff
@@ -134,6 +135,56 @@ That’s our lexer definition it has just token channel which are going to be us
+
+#### Bytecode
+
+`Bytecode` is an output generated by `compiler`. Through `vm` executing this code, we can get result of the source code.
+
+```go
+type Bytecode struct {
+   RawByte []byte
+   AsmCode []string
+   Abi     abi.Abi
+}
+```
+
+This is our `Bytecode` structure. It has 3 fields. `RawByte` is a byte which has information of how to execute the program. And `RawByte` consists of hexadecimal code. `AsmCode` is a collection of assemble codes which is more readable to human than bytes. `Abi` is an interface needed to user for calling the functions.
+
+<p align="center"><img src="../image/bytecode-structure.jpg" width="600px" height="40px"></p>
+
+The raw bytecode is structed like above. `VM Memory Setting code` sets memory size of the `VM`. `Function Jumper` could find the position of each function. And, the functions of contract would be followed by the `Function Jumper`. Each function bytecode has the `function selector`, parameters, and logic.
+
+[Function Jumper](#function-jumper) and [Function Selector](#function-selector) will be explained later.
+
+#### AST & Compile
+
+<p align="center"><img src="../image/ast-architecture.jpg" width="470px" height="340px"></p>
+
+The root of AST is a `contract`. `Contract` has some `functions`. And, a `function` has some `statements`.
+
+`CompileContract()` function compiles the contract. It compiles every function and adds `VM memory setting code` and `jump selector`. `compileFunction()` function compiles a function in the contract. It is called recursively and compiles each function. `compileStatement()` function compiles a statement in the function. It is called recursively and compiles each statement.
+
+#### <a name="function-selector">Function Selector</a>
+
+Each function should have identifiers to distinguish each other. The `function selector` is an identifier to distinguish each function. It is generated by Keccak algorithm (SHA-3). This algorithm generates a bytes with function name and parameter type. And the first 4 bytes of those bytes are the `function selector`. For example, the `function selector` of `function foo(int a) bool` is `0x4ff9f498`. The keccak algorithm receives `"foo(int)"` and generates some bytes. The first 4 bytes of those are `0x4ff9f598` and this is the `function selector`.
+
+#### <a name="function-jumper">Function Jumper</a>
+
+The `function jumper` includes a logic to find a function to call. If an user calls `function a()`, program counter points `function jumper`. Because of the static size of `function jumper`, the program could know the position of `function jumper`. This finds the `function selector` corresponding with `function a()`. Then, returns the bytes of `function a()`'s `function selector`. Finally, `VM` could know the position `function a()` and execute it.
```

Comment:
Not because of the static size of `function jumper`, but because of the static location of `function jumper`